File: 0007-reverse-integer/solution.cpp
class Solution {
public:
    int reverse(int x) {
       int dig;
        long long rev=0;
        while(x!=0){
            dig = x%10;
            rev = (rev*10)+dig;
            x=x/10;
        }
        if(rev < INT_MIN||rev > INT_MAX)
        {
            return 0;
        }
        else
        return rev;
    }
};

File: 0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        if x >=0:
            rev = int(str(x)[::-1])
        else:
            rev = -1 * int(str(x)[1:][::-1])
        if rev < -2**31 or rev > 2**31-1:
            rev = 0
        
        return rev

        

